projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -1145,21 +1162,23 @@ export class IgxTimePickerComponent implements
     public openDialog(timePicker: IgxTimePickerComponent = this): void {
         if (this.mode === InteractionMode.Dialog) {
             this.collapsed = false;
+            const dialogSettings = this.overlaySettings ? this.overlaySettings : this._dialogOverlaySettings;
             if (this.outlet) {
-                this._dialogOverlaySettings.outlet = this.outlet;
+                dialogSettings.outlet = this.outlet;
             }
-            this._overlayId = this.overlayService.attach(this.container, this._dialogOverlaySettings);
+            this._overlayId = this.overlayService.attach(this.container, dialogSettings);
             this.overlayService.show(this._overlayId);
         }
 
         if (this.mode === InteractionMode.DropDown) {
             if (this.collapsed) {
                 this.collapsed = false;
+                const dropdownSettings = this.overlaySettings ? this.overlaySettings : this._dropDownOverlaySettings;
```

Comment:
`this.overlaySettings ? this.overlaySettings : this._dropDownOverlaySettings;` could be simplified to `this.overlaySettings || this._dropDownOverlaySettings;`.